src/Raven.Server/Documents/Indexes/MapReduce/OutputToCollection/OutputReduceToCollectionCommand.cs

```diff
@@ -400,54 +400,41 @@ public void Execute(DocumentsOperationContext context, Transaction indexWriteTra
                 {
                     using (var existingReferenceDocument = _database.DocumentsStorage.Get(context, referencesOfReduceOutput.Key))
                     {
-                        if (existingReferenceDocument == null)
+                        var uniqueIds = referencesOfReduceOutput.Value;
+
+                        if (existingReferenceDocument != null)
                         {
-                            var referenceDoc = new DynamicJsonValue
-                            {
-                                [nameof(OutputReduceToCollectionReference.ReduceOutputs)] = new DynamicJsonArray(referencesOfReduceOutput.Value),
-                                [Constants.Documents.Metadata.Key] = new DynamicJsonValue
-                                {
-                                    [Constants.Documents.Metadata.Collection] = $"{_outputReduceToCollection}/References"
-                                }
-                            };
-
-                            using (var referenceJson = context.ReadObject(referenceDoc, "reference-of-reduce-output", BlittableJsonDocumentBuilder.UsageMode.ToDisk))
+                            if (existingReferenceDocument.Data.TryGet(nameof(OutputReduceToCollectionReference.ReduceOutputs), out BlittableJsonReaderArray existingIds) == false)
+                                ThrowIdsPropertyNotFound(existingReferenceDocument.Id);
+
+                            foreach (object id in existingIds)
                             {
-                                PutReferenceDocument(referencesOfReduceOutput, referenceJson);
+                                uniqueIds.Add(id.ToString());
                             }
                         }
-                        else
-                        {
-                            if (existingReferenceDocument.Data.TryGet(nameof(OutputReduceToCollectionReference.ReduceOutputs), out BlittableJsonReaderArray ids) == false)
-                                ThrowIdsPropertyNotFound(existingReferenceDocument.Id);
-
-                            if (ids.Modifications == null)
-                                ids.Modifications = new DynamicJsonArray();
 
-                            foreach (var reduceOutputId in referencesOfReduceOutput.Value)
+                        var referenceDoc = new DynamicJsonValue
+                        {
+                            [nameof(OutputReduceToCollectionReference.ReduceOutputs)] = new DynamicJsonArray(referencesOfReduceOutput.Value),
```

Comment:
It should be: `uniqueIds` here

Reply:
and need a test for this

Reply:
Thanks for catching this. Regarding a test - I saw it on your end but I'm not actually sure how it  happened.

Reply:
Fixed in https://github.com/ravendb/ravendb/pull/10055/commits/017ca971f6456ff43e7a30c71c982c7203309dd7

Reply:
Test added in https://github.com/ravendb/ravendb/pull/10055/commits/15285e3e932b39e3fbd946928b19205f74ca34ad